makefile

```diff
@@ -208,9 +211,9 @@ $(buildDir)/make-tarball:cmd/make-tarball/make-tarball.go
 	@GOOS="" GOARCH="" $(gobin) build -o $@ $<
 	@echo $(gobin) build -o $@ $<
 
-
-dist-staging: STAGING_ONLY := 1
-dist-staging:$(buildDir)/dist.tar.gz
+dist-staging: export STAGING_ONLY := 1
+dist-staging:
```

Comment:
I think this should be
```
dist-staging: dist
```

instead of
```
dist-staging
    make dist
```

The idea is that you define dist as a prerequisite of dist-staging and let make do the rest.

Reply:
This will not work due to the target expansion problem with `$(clientBuildDir)/%/evergreen`.  The first time make is called, there is an initial evaluation without `STAGING_ONLY := 1`, which gets picked up by the `%` expansion and causes the problematic behavior. Only when you call an entirely new make with the exported variable does clientBinaries expand to the desired values in time for the `$(clientBuildDir)/%/evergreen` target to catch it.